pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -42,6 +43,40 @@ func (r *Reconciler) appendEnvFrom(envList []corev1.EnvFromSource, secret string
 	})
 }
 
+// appendVolumeMounts append volume mounts pointing to volumes created using secret
+func (r *Reconciler) appendVolumeMounts(vmList []corev1.VolumeMount, volumeName string) []corev1.VolumeMount {
+	for _, vm := range vmList {
+		if vm.Name == volumeName {
+			// volume name already referenced
+			return vmList
+		}
+	}
+
+	return append(vmList, corev1.VolumeMount{
+		Name:      volumeName,
+		MountPath: "/" + volumeName,
+	})
+}
+
+// appendVolumes append volumes
+func (r *Reconciler) appendVolumes(volumeList []corev1.Volume, volumeName, secretName string) []corev1.Volume {
+	for _, vm := range volumeList {
+		if vm.Name == volumeName {
+			// volume name already referenced
+			return volumeList
+		}
+	}
+
+	return append(volumeList, corev1.Volume{
+		Name: volumeName,
+		VolumeSource: corev1.VolumeSource{
```

Comment:
What is the corresponding yaml/json structure for the `VolumeSource` , and why do we need it?

Reply:
The YAML structure will be like this:
```
  volumes:
    - name: secret-volume
      secret:
        secretName: test-secret
```
This volume is referred from the volumeMount.

Reply:
I have added items. 935a50dc49bc80180f09481339f56367567c479a